script.js

```diff
@@ -0,0 +1,69 @@
+const addbookBtn = document.querySelector('#add-book');
+const bookTitle = document.querySelector('#title');
+const bookAuthor = document.querySelector('#author');
+const bookList = JSON.parse(localStorage.getItem('bookShop')) || [];
+
+const addBook = (bookTitle, bookAuthor) => {
+  const book = {};
+  book.title = bookTitle.value;
+  book.author = bookAuthor.value;
+  bookList.push(book);
+
+  bookTitle.value = '';
+  bookAuthor.value = '';
+};
+
+const displayBook = (book) => {
+  const { title, author } = book;
+  const bookContainer = document.createElement('div');
+  bookContainer.className = 'book';
+  bookContainer.innerHTML = `<span class='title'>${title}</span><br>
+    <span class='by'>by</span><br>
+    <span class='author'>${author} </span><br>
+    <button class='delete'>Remove</button>
+    <hr>`;
+  return bookContainer;
+};
+
+const displayBooks = () => {
+  const currentBooks = document.querySelector('#books-list');
+  const libraryContainer = document.createElement('div');
+
+  bookList.forEach((book) => {
+    libraryContainer.appendChild(displayBook(book));
+    currentBooks.innerHTML = libraryContainer.innerHTML;
+  });
+};
+
+addbookBtn.addEventListener('click', (e) => {
+  e.preventDefault();
+  if (bookTitle.value === null || bookAuthor.value === null) {
+    return;
+  }
+  addBook(bookTitle, bookAuthor);
+  localStorage.setItem('bookShop', JSON.stringify(bookList));
+
+  displayBooks();
+});
+
+window.onload = displayBooks();
+
+// Remove Book
+const removeBook = (button) => {
+  const bookList = JSON.parse(localStorage.getItem('bookShop'));
+  const parentDiv = button.parentNode;
+  const myTitle = parentDiv.querySelector('.title').textContent;
+  const myAuthor = parentDiv.querySelector('.author').textContent;
+  const booksLeft = bookList.filter(
+    (book) => book.title !== myTitle && book.author !== myAuthor,
+  );
```

Comment:
- [ ] Great job so far, as the previous reviewer pointed out when you try to remove a book and then add another one, all of the books are displayed, even the removed one. Kindly fix this behaviour. Please make sure when a user removes a book the object in local storage gets updated and the displayBook function uses the updated list.

Reply:
Ok I have checked the functionality of the project and its working from my computer. When I add a book to the list and remove a book from the list, it will store the updated list in the local storage and display it. i really dont understand why its not responding as you said but its working well from my computer and my patners own too.

Reply:
Hi @Mikelobi did you guys push all the changes?

Reply:
> * [ ]  Great job so far, as the previous reviewer pointed out when you try to remove a book and then add another one, all of the books are displayed, even the removed one. Kindly fix this behaviour. Please make sure when a user removes a book the object in local storage gets updated and the displayBook function uses the updated list.

Please help me with a clip video so i can see the way its working from your end.

Reply:
@Wes-Isaac. Please help me with a clip video so i can see the way its working from your end.

Reply:
> Hi @Mikelobi did you guys push all the changes?

yes we did.

Reply:
Hi @Wes-Isaac, yes we pushed all changes

Reply:
I checked again locally still has a problem, did you understand what the previous reviewer asked?

Reply:
> 

OK Please wait for me

Reply:
> 

OK Please wait for me

Reply:
https://www.loom.com/share/e70631029710474b9f06dcd68509a6b2 Please check this out

Reply:
![](https://www.loom.com/share/db345613564744a2bba8e5f1d04c8e6d)

Reply:
> https://www.loom.com/share/e70631029710474b9f06dcd68509a6b2 Please check this out

Please check this link. https://www.loom.com/share/db345613564744a2bba8e5f1d04c8e6d

Reply:
@Wes-Isaac. Please check this link. https://www.loom.com/share/db345613564744a2bba8e5f1d04c8e6d.

Reply:
Hi @Mikelobi I have watched the video and noticed that every time you remove you were refreshing the browser, but if you don't refresh at all and check you will see what the previous reviewer and myself pointed out.

Reply:
When I did the same thing as you did (refreshing after removing) it works but the app should work like that without needing a refresh every time.

Reply:
Hi @Wes-Isaac It is not a must that if i click on remove button without refreshing it will not display the updated list. let me show you another video without reloading the page.

Reply:
ok @Wes-Isaac let me check again. I used to refresh the page